knotx-server/src/main/java/io/knotx/server/configuration/KnotxCSRFConfig.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.server.configuration;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.CSRFHandler;
+
+
+public class KnotxCSRFConfig {
+
+  private long DEFAULT_TIMEOUT = 30 * 60 * 1000; // 30 minutes
+
+  private static final String DEFAULT_SECRET = "a31DDUam$D{h{x5*ah2j0,EQhkA`Su";
+
+  private String cookieName;
+
+  private String cookiePath;
+
+  private String headerName;
+
+  private String secret;
+
+  private long timeout;
+
+  public KnotxCSRFConfig(JsonObject config) {
```

Comment:
This class is really simple, but maybe we can use `@DataObject(generateConverter = true)` here?
Does it support default values?

Reply:
I'd rather not go with data object generator now. It will make a incosistency. Ideally, all the configuration objects should be treated as DataObject - that will result in a great support for documentation purposes, but I it's something we should do globally for all configuration objects.